src/hotspot/share/jeandle/jeandleAbstractInterpreter.cpp

```diff
@@ -195,6 +195,42 @@ void JeandleVMState::store(BasicType type, int index, llvm::Value* value) {
   }
 }
 
+
+llvm::SmallVector<llvm::Value*> JeandleVMState::deopt_args(llvm::IRBuilder<>& builder) {
+  llvm::SmallVector<llvm::Value*> args;
+  // |--- loc ---|--- stk ---|--- arg ---|--- mon ---|--- scl ---|
+  /* TODO: monitor and scalar */
+  for (size_t i=0; i < _locals.size(); i++) {
+    if (!_locals[i].is_null()) {
+      uint64_t encode = DeoptValueEncode(i, DeoptValueEncode::LocalType, _locals[i].computational_type()).encode();
+      args.push_back(builder.getInt64(encode));
+      args.push_back(_locals[i].value());
+    } else {
+      // replace with {T_ILLEGAL, 0}
+      uint64_t encode = DeoptValueEncode(i, DeoptValueEncode::LocalType, T_ILLEGAL).encode();
+      args.push_back(builder.getInt64(encode));
+      args.push_back(builder.getInt32(0));
+    }
```

Comment:
Maybe this logic can be extracted as a function, which can be used by stack deopt value generating below, too.